Control/Parallel/Strategies.hs

```diff
@@ -432,19 +432,33 @@ rpar  x = case (par# x) of { _ -> Done x }
 #endif
 {-# INLINE rpar  #-}
 
--- | instead of saying @rpar `dot` strat@, you can say
--- @rparWith strat@.  Compared to 'rpar', 'rparWith'
+-- | Perform a computation in parallel using a strategy.
 --
---  * does not exit the `Eval` monad
---
---  * does not have a built-in `rseq`, so for example `rparWith r0`
---    behaves as you might expect (it is a strategy that creates a
---    spark that does no evaluation).
+-- @
+-- rparWith strat x
+-- @
 --
+-- will spark a thread to perform @strat x@ in parallel. Note
+-- that @rparWith strat@ is /not/ the same as @rpar `dot` strat@.
+-- Specifically, @rpar `dot` strat@ always produces a value in
+-- WHNF, while @rparWith strat@ need not.
 --
+-- > rparWith r0 = r0
+-- > rparWith rpar = rpar
```

Comment:
The LHS creates two sparks, I'm not sure I'd call them equal. It depends what equality you're using, but as I argued elsewhere the only sensible notions of equality here are ones that care about sparks.

Reply:
`rparWith rpar` is actually a quite weird thing: it creates a spark, who creates a spark, who evaluates to WHNF.

However one can probably replace that law with `rparWith rseq = rpar`.

Reply:
How does one replace the other? What we probably actually want is some sort of equivalence relation that equates "essentially equivalent" things. `rpar x >>= rseq` is effectively the same as `rseq x`, but I don't know how to express that.

Reply:
@simonmar, my intuition suggests that we're never going to find all the relations (`Eval` is a very large structure), but that we should document some of the more useful ones. It's easy to *think* you're getting more parallelism than you are when composing things. Speaking of relations,

```haskell
runEval (m >>= rpar) = runEval m

rpar x >>= \x' -> rpar y >>= f  =  rpar y >>= \y' -> rpar x >>= f
-- when x' is not free in y and y' is not free in x.
```